src/transformState.js

```diff
@@ -5,7 +5,19 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const element of actions) {
+    if (element.type === 'addProperties') {
+      Object.assign(state, element.extraData);
+    } else if (element.type === 'removeProperties') {
+      for (const removeKeys of element.keysToRemove) {
+        delete state[removeKeys];
+      }
+    } else if (element.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+  }
```

Comment:
You have a lot of nesting `if`'s here, it'd be better to use switch/case construction here